### ISSUE TITLE
Next.js 13 server components type error

### ISSUE THREAD

**Issue by transitive-bullshit** [timolins/react-hot-toast#232]
Using a beta of Next.js 13, running `next dev` with a `<Toaster />` in a server component results in:

```
TypeError: (0 , k.useState) is not a function or its return value is not iterable
```

I'm guessing this has something to do with `react-hot-toast` being used in a server component. If I move the `Toaster component into an explicit client component file, it works as expected:

```ts
'use client'

import { Toaster } from 'react-hot-toast'

export function Client() {
  return <Toaster position='top-right' />
}
```

Some notes on env (though I don't think it matters):
- react-hot-toast 2.4.0
- next 13.0.2
- react 18.2.0
- react-dom 18.2.0
- node.js 16.18.0
- macOS 12.6

**Comment by pellyadolfo** [comment]
Finally this worked for me with Next.js

https://stackoverflow.com/a/72815922/2517585

**Comment by Tomekmularczyk** [comment]
I'm getting this (Nextjs@13.1.6, react-hot-toast@2.4.0):

```
error - ./node_modules/react-hot-toast/dist/index.mjs
ReactServerComponentsError:

You're importing a component that needs useEffect. It only works in a Client Component but none of its parents are marked with "use client", so they're Server Components by default.
```

**Comment by baba43** [comment]
Well, turning the whole app to a client component is actually not a fix, it is a very ugly workaround.

Is there smarter solution out there?

**Comment by timolins** [comment]
This is now fixed with [`2.4.1`](https://github.com/timolins/react-hot-toast/releases/tag/v2.4.1), by including "use client" inside the package. 🙏
